internal/dev_server/manifesto.md

```diff
@@ -0,0 +1,17 @@
+# Dev Server Vision Doc
+
+This doc lays out our vision for what `dev-server` is supposed to do, and why this is the case.
+
+## Purpose of the Dev Server
+
+First and foremost, `dev-server` is meant as a replacement for local dev and CI environments. Other use cases may exist (and don't hestitate to ask if you think you have one!), but since a `dev-server` environment lacks any targeting, it is best suited to situations where there is only 1 user for the whole environment.
+
+We implemented it this way in order to solve a problem common both internally and amongst our userbase where an environment would get created per developer on a project as well as per CI build. As you could expect, this becomes exponentially harder to maintain as the number of environments scales both technically and from a process perspective - something has to clear out those environments after all!
+
+Another intentional design decision was the choice not to allow the creation of local-only flags. A running `dev-server` can only pull from a real environment. From there, you can override the served value for your local environment.
```

Comment:
Rewrite this without mentioning local-only flags. That's a feature we've discussed internally, but isn't something a reader of this doc will know about. I think you want to frame it as "flags must first exist in the source environment" or similar.